src/moscot/problems/time/_lineage.py

```diff
@@ -249,15 +253,20 @@ def prepare(
         if not len(lineage_attr) and ("cost_matrices" not in self.adata.obsp):
             raise KeyError("Unable to find cost matrices in `adata.obsp['cost_matrices']`.")
 
-        lineage_attr = dict(lineage_attr)
-        lineage_attr.setdefault("attr", "obsp")
-        lineage_attr.setdefault("key", "cost_matrices")
-        lineage_attr.setdefault("cost", "custom")
-        lineage_attr.setdefault("tag", "cost_matrix")
-
         x = y = lineage_attr
 
         xy, kwargs = handle_joint_attr(joint_attr, kwargs)
+        xy, x, y = handle_cost(xy=xy, x=x, y=y, cost=cost, cost_kwargs=cost_kwargs)
+
+        x.setdefault("attr", "obsp")
+        x.setdefault("key", "cost_matrices")
+        x.setdefault("cost", "custom")
+        x.setdefault("tag", "cost_matrix")
+        y.setdefault("attr", "obsp")
```

Comment:
Please add a space between the last `x.setdefault` and the first set `y.default`.